proto/serving.proto

```diff
@@ -148,3 +149,28 @@ message ResourceIdRequest {
 message ResourceLocation {
   string location = 1;
 }
+
+message TrainingTestSplitRequest {
+  TrainingDataID id = 1;
+  Model model = 2;
+  float test_size = 3;
+  float train_size = 4;
+  bool shuffle = 5;
+  int32 random_state = 6;  // Seed for shuffling, if shuffle is true
+  RequestType request_type = 7;  // NEW: Specify the type of data being requested
+}
+
+enum RequestType {
+  INITIALIZE = 0;
+  TRAINING = 1;  // Client is requesting training data
+  TEST = 2;      // Client is requesting test data
+}
+
+message TrainingTestSplitResponse {
```

Comment:
wasn't the happiest with this,

in retrospect and initalized probably isn't necessary.
iterator done means first iterator is done and it's a response we send so the client knows to finish iterating on the first iterator but keep the stream open; i'll add a comment